app/src/main/java/com/example/dispatchbuddy/presentation/ui/authentication/SmsVerificationFragment.kt

```diff
@@ -25,26 +25,29 @@ import com.example.dispatchbuddy.common.ViewExtensions.hideKeyBoard
 import com.example.dispatchbuddy.common.ViewExtensions.hideView
 import com.example.dispatchbuddy.common.ViewExtensions.showShortSnackBar
 import com.example.dispatchbuddy.common.ViewExtensions.showView
+import com.example.dispatchbuddy.common.preferences.Preferences
 import com.example.dispatchbuddy.data.remote.dto.models.VerifyUser
 import com.example.dispatchbuddy.databinding.FragmentSmsVerificationBinding
 import com.example.dispatchbuddy.presentation.ui.authentication.viewmodel.VerificationViewModel
 import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.coroutines.launch
+import javax.inject.Inject
 
 
 @AndroidEntryPoint
 class SmsVerificationFragment : Fragment() {
     private var _binding: FragmentSmsVerificationBinding? = null
     private val binding get() = _binding!!
     val args: SmsVerificationFragmentArgs by navArgs()
-    val TAG = "SmsVerificationFragment"
     private val verificationViewModel: VerificationViewModel by viewModels()
+    @Inject
+    lateinit var preferences: Preferences
 
     override fun onCreate(@Nullable savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val callback: OnBackPressedCallback = object : OnBackPressedCallback(true /* enabled by default */) {
             override fun handleOnBackPressed() {
-                findNavController().popBackStack()
+                findNavController().navigate(R.id.action_smsVerificationFragment_to_emailForChangePasswordFragment)
             }
```

Comment:
There is also a flow from registration screen. If the user comes to this fragment from registration screen he should not see emailForChangePasswordFragment when he presses back button

Reply:
Oh, right. I'll change that popBackStack() then.